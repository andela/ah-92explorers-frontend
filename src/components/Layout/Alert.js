/* eslint-disable react/forbid-prop-types */
/* eslint-disable react/require-default-props */
import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

export const Alert = ({ alerts }) => (
  alerts !== undefined
    && alerts !== null
    && alerts.length > 0
    && alerts.map(alert => (
      <div key={alert.id} className={`alert alert-${alert.alertType}`}>
        {alert.msg}
      </div>
    ))
);

Alert.propTypes = {
  alerts: PropTypes.array,
};

const mapStateToProps = state => ({
  alerts: state.alert,
});

export default connect(mapStateToProps)(Alert);
