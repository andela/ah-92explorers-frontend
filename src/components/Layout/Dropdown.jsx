/* eslint-disable jsx-a11y/no-static-element-interactions */
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { logout } from '../../redux/actions/actionCreators/login';
import '../../assets/css/dropdown.css';

export class Dropdown extends Component {
  constructor(props) {
    super(props);
    this.state = {
      displayMenu: false,
    };
  }

  showDropdownMenu = (event) => {
    event.preventDefault();
    this.setState({ displayMenu: true }, () => {
      document.addEventListener('click', this.hideDropdownMenu);
    });
  }

  hideDropdownMenu = () => {
    this.setState({ displayMenu: false }, () => {
      document.removeEventListener('click', this.hideDropdownMenu);
    });
  }

  logout = (e) => {
    e.preventDefault();
    this.props.logout();
  }

  render() {
    return (
      <div className="menu-dropdown" style={{ width: '100px' }}>
        <div className="user-caret" onClick={this.showDropdownMenu} onKeyDown={this.handleClick}>
          <span className="fa fa-caret-down" />
        </div>

        { this.state.displayMenu ? (
          <ul>
            <li><Link to="/profile">Profile</Link></li>
            <hr />
            <li><button type="button" onClick={this.logout} className="signout">Sign out</button></li>
          </ul>
        )
          : (
            null
          )}

      </div>

    );
  }
}

Dropdown.propTypes = {
  logout: PropTypes.func.isRequired,
};

function mapStateToProps(state) {
  return {
    login: state.login,
  };
}


export default connect(mapStateToProps, { logout })(Dropdown);
