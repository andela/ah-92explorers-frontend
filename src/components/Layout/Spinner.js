import React, { Fragment } from 'react';
import spinner from '../../assets/images/loading.gif';

const Spinner = () => (
  <Fragment>
    <img
      src={spinner}
      style={{ width: '70px', margin: 'auto', display: 'block' }}
      alt="loading"
    />
  </Fragment>
);

export default Spinner;
