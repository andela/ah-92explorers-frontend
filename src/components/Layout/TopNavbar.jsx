import React, { Component, Fragment } from 'react';
import { Link } from 'react-router-dom';
import logo from '../../assets/icons/logo.png';

class TopNavbar extends Component {
  render() {
    return (
      <Fragment>
        <div className="navbar">
          <img src={logo} alt="" className="logo dash-logo" />
          <div className="restOfNav">
            <ul>
              <li>
                {' '}
                <Link to="/login">Login</Link>
              </li>
              <li>
                {' '}
                <Link to="/signup">Sign up</Link>
              </li>
            </ul>
          </div>
        </div>

      </Fragment>
    );
  }
}

export default TopNavbar;
