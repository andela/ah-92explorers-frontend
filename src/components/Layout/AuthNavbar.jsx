import React from 'react';
import { Link } from 'react-router-dom';
import Dropdown from './Dropdown.jsx';


const AuthNavbar = () => (
  <nav>
    <div className="firstNavbar">
      <button className="createArt" type="button">
        <Link to="./createArticle.html">
          <img src={require('../../assets/icons/edit.svg')} alt="" className="articleBtn" />
            Write Article
        </Link>
      </button>
      <img src={require('../../assets/icons/logo.png')} alt="" className="logo" />
      <input className="search-bar" type="text" name="search" placeholder="Title, author, tag, keyword ..." />
      <div>
        <img src={require('../../assets/icons/notification.svg')} alt="" className="navIcons notifications disappear" />
        <span className="notif-count">6</span>
      </div>
      <div className="notifications-drop">
        <ul>
          <li className="title">
hillary liked your article
            <span>unread</span>
          </li>
          <li className="title">
hillary liked your article
            <span>unread</span>
          </li>
          <li className="title">
hillary liked your article
            <span>unread</span>
          </li>
          <li className="title">
hillary liked your article
            <span>unread</span>
          </li>
          <li className="title">
hillary liked your article
            <span>unread</span>
          </li>
        </ul>
      </div>
      <img src={require('../../assets/icons/man.svg')} alt="" className="manIcon" />
      <Dropdown />
    </div>
  </nav>
);

export default AuthNavbar;
