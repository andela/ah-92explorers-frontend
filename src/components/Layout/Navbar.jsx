/* eslint-disable global-require */
import React from 'react';
import { Link } from 'react-router-dom';
import Dropdown from './Dropdown.jsx';

const Navbar = ({ token, username, avatar }) => {
  if (!token || token === undefined) {
    return (
      <nav className="feedNavbar" style={{ backgroundColor: '#fffff' }}>
        <div className="firstNavbar">
          <a href="/">
            <img src={require('../../assets/icons/logo.png')} alt="" className="logoNavbar" />
          </a>
          <div className="notifications-drop">
            <ul>
              <li className="title">
            hillary liked your article
                <span>unread</span>
              </li>
              <li className="title">
            hillary liked your article
                <span>unread</span>
              </li>
              <li className="title">
            hillary liked your article
                <span>unread</span>
              </li>
              <li className="title">
            hillary liked your article
                <span>unread</span>
              </li>
              <li className="title">
            hillary liked your article
                <span>unread</span>
              </li>
            </ul>
          </div>
          <button className="signupBtnNav disappear dBtnOne" type="button"><Link to="/signup">Sign Up</Link></button>
          <button className="signinBtnNav disappear" type="button"><Link to="/login">Sign In</Link></button>

        </div>
        <div className="secondNavbar">
          <div className="listBtns disappearBtn">
            <button className="signupBtnNav" type="button"><Link to="/signup">Sign Up</Link></button>
            <button className="signinBtnNav" type="button"><Link to="/login">Sign In</Link></button>
          </div>
        </div>
      </nav>
    );
  }
  return (
    <nav className="feedNavbar feedLoggedIn" style={{ backgroundColor: '#fffff' }}>
      <div className="firstNavbar">
        <img src={require('../../assets/icons/logo.png')} alt="" className="logoNavbar" />
        <div className="notifications-drop">
          <ul>
            <li className="title">
            hillary liked your article
              <span>unread</span>
            </li>
            <li className="title">
            hillary liked your article
              <span>unread</span>
            </li>
            <li className="title">
            hillary liked your article
              <span>unread</span>
            </li>
            <li className="title">
            hillary liked your article
              <span>unread</span>
            </li>
            <li className="title">
            hillary liked your article
              <span>unread</span>
            </li>
          </ul>
        </div>
        <div className="userStatus">
          <img src={avatar} alt="" className="avatarFeed" />
          <span id="userStatusName">{username}</span>
          <Dropdown />
        </div>
      </div>
    </nav>
  );
};

export default Navbar;
