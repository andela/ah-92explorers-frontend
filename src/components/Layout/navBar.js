import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import dotenv from 'dotenv';
import logo from '../../assets/icons/logo.png';
import search from '../../assets/icons/search.svg';
import notification from '../../assets/icons/notification.svg';
import Dropdown from './Dropdown.jsx';

dotenv.config();

class Navbar extends Component {
  render() {
    const { profile } = this.props;
    return (
      <Fragment>
        <div className="navbar">
          <a href="/"><img src={logo} alt="" className="logo dash-logo" /></a>
          <ul>
            <li><Link to="/article">Create Article</Link></li>
            <li><Link to="/profile">Profile</Link></li>
          </ul>
          <div className="restOfNav">
            <img src={search} alt="" className="navIcons navIconColor" />
            <img src={notification} alt="" className="navIcons navIconColor" />
            <img src={(profile && profile.image) || process.env.DEFAULT_IMAGE} alt="" className="navIcons manIcon" />
            <Dropdown />
          </div>
        </div>

      </Fragment>
    );
  }
}

function mapStateToProps(state) {
  return {
    profile: state.profile.profile,
  };
}


export default connect(mapStateToProps)(Navbar);
