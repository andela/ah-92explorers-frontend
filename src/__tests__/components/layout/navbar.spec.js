/* eslint-disable import/no-unresolved */
import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { Navbar } from '../../../components/Layout/navBar';

const mockStore = configureMockStore();
const store = mockStore({});
describe('<Navbar />', () => {
  const props = {
    notifications: {
      notifications: {
          allNotification: []
      },
    },
    getNotifications: jest.fn(),
    readNotification: jest.fn(),
  }
  const wrapper = shallow(
      <Navbar {...props}/>
  );
  it('should render Navbar component', () => {
    const li = wrapper.find('ListGroupItem').first();
    expect(wrapper).toMatchSnapshot();
  });
  
  it('test update state onchange', () => {
    wrapper.instance().componentDidMount()
    expect(wrapper.instance().props.getNotifications).toBeCalled();
  });

  it('shoudl handle on comment', () => {
    const component = wrapper.instance();
    component.onclickNotif({ target: { getAttribute: jest.fn()}});
    component.refresh();
    component.onHandleClick();
    component.handleMouseLeave();
    component.toggle2()
    component.viewmore();
    expect(component).toBeDefined();
  });
});
