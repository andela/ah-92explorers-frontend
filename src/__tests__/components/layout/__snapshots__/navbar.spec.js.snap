// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should render Navbar component 1`] = `
<nav>
  <div
    className="firstNavbar"
  >
    <img
      alt=""
      className="logo"
      src="test-file-sdub"
    />
    <input
      className="search-bar"
      name="search"
      placeholder="Title, author, tag, keyword ..."
      type="text"
    />
    <div
      className="notifications-drop"
    >
      <ul>
        <li
          className="title"
        >
          hillary liked your article
          <span>
            unread
          </span>
        </li>
        <li
          className="title"
        >
          hillary liked your article
          <span>
            unread
          </span>
        </li>
        <li
          className="title"
        >
          hillary liked your article
          <span>
            unread
          </span>
        </li>
        <li
          className="title"
        >
          hillary liked your article
          <span>
            unread
          </span>
        </li>
        <li
          className="title"
        >
          hillary liked your article
          <span>
            unread
          </span>
        </li>
      </ul>
    </div>
    <button
      className="signup"
      type="button"
    >
      <Link
        to="/"
      >
        Sign Up
      </Link>
    </button>
    <button
      className="signin"
      type="button"
    >
      <Link
        to="/login"
      >
        Sign In
      </Link>
    </button>
  </div>
  <div
    className="secondNavbar"
  >
    <ul>
      <li>
        Popular
      </li>
      <li>
        Popular
      </li>
      <li>
        Popular
      </li>
      <li>
        Popular
      </li>
      <li>
        Popular
      </li>
      <li>
        Popular
      </li>
      <li>
        Popular
      </li>
      <li>
        Popular
      </li>
      <li>
        Popular
      </li>
      <li>
        Popular
      </li>
    </ul>
  </div>
</nav>
`;
